src/DropDownMenu/DropDownMenu.jsx

```diff
@@ -0,0 +1,379 @@
+            onTouchTap={this._onMenuItemTouchTap.bind(this, idx, item[valueMember])} />
+        ))
+      : React.Children.map(children, child => {
+        const clone = React.cloneElement(child, {
+          onTouchTap: this._onMenuItemTouchTap.bind(this, index, child.props.value),
+        }, child.props.children);
+        index += 1;
+        return clone;
+      });
+
+    let popoverStyle;
+    if (anchorEl && !autoWidth) {
+      popoverStyle = {width: anchorEl.clientWidth};
+    }
+
+    return (
+      <div
+        {...other}
+        ref="root"
+        className={className}
+        style={this.prepareStyles(styles.root, open && styles.rootWhenOpen, style)}
+      >
+        <ClearFix style={this.mergeStyles(styles.control)} onTouchTap={this._onControlTouchTap}>
+          <div style={this.prepareStyles(styles.label, open && styles.labelWhenOpen, labelStyle)}>
+            {displayValue}
+          </div>
+          <DropDownArrow style={this.mergeStyles(styles.icon, iconStyle)}/>
+          <div style={this.prepareStyles(styles.underline, underlineStyle)}/>
+        </ClearFix>
+        <Popover
+          anchorOrigin={{horizontal: 'left', vertical: 'top'}}
+          anchorEl={anchorEl}
+          style={popoverStyle}
+          animation={PopoverAnimationFromTop}
+          open={open}
+          onRequestClose={this._onMenuRequestClose} >
+          <Menu
+            maxHeight={maxHeight}
+            desktop={true}
+            value={value}
+            style={menuStyle}
+            >
+            {menuItemElements}
+          </Menu>
+        </Popover>
+      </div>
+    );
+  },
+
+  _testDeprecations() {
+    warning(!this.props.hasOwnProperty('displayMember'),
+      'displayMember will be removed in favor of composability. refer to the documentation for more information');
+
+    warning(!this.props.hasOwnProperty('labelMember'),
+      'labelMember will be removed in favor of composability. refer to the documentation for more information');
+
+    warning(!this.props.hasOwnProperty('labelStyle'),
+      'labelStyle will be removed with labelMember.');
+
+    warning(!this.props.hasOwnProperty('menuItems'),
+      'menuItems will be removed in favor of composability. refer to the documentation for more information');
+
+    warning(!this.props.hasOwnProperty('selectedIndex'),
+      'selectedIndex will be removed. use value instead to control the component.');
+
+    warning(!this.props.hasOwnProperty('valueLink'),
+      'valueLink will be removed. use value and onChange');
+
+    warning(!this.props.hasOwnProperty('valueMember'),
+      'valueMember will be removed in favor of composability. refer to the documentation for more information');
+  },
+
+  _setWidth() {
+    const el = ReactDOM.findDOMNode(this);
+    if (!this.props.style || !this.props.style.hasOwnProperty('width')) {
+      el.style.width = 'auto';
+    }
+  },
+
+  _setSelectedIndex(index) {
+    if (process.env.NODE_ENV !== 'production') {
```

Comment:
I think we can remove this production check since the warning package does it for us.